sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2367,6 +2367,21 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         case agg: AggregateFunction =>
           // Note: PythonUDAF does not support these advanced clauses.
           if (agg.isInstanceOf[PythonUDAF]) checkUnsupportedAggregateClause(agg, u)
+          val newAgg = agg match {
+            case factory: InverseDistributionFactory if u.sortOrder.isDefined =>
+              if (u.isDistinct) {
+                throw QueryCompilationErrors.distinctInverseDistributionFunctionUnsupportedError(
+                  factory.prettyName)
+              }
+              factory.createInverseDistributionFunction(u.sortOrder.get)
+            case factory: InverseDistributionFactory =>
+              throw QueryCompilationErrors.inverseDistributionFunctionMissingWithinGroupError(
+                factory.prettyName)
+            case other if u.sortOrder.isDefined =>
+              throw QueryCompilationErrors.unsupportedInverseDistributionFunctionError(
```

Comment:
It only covers unsupported agg func. We should cover all cases in `validateFunction`

Reply:
maybe we can add an extra case match at the beginning
```
case _ if !func.isInstanceOf[InverseDistributionFactory] && u.sortOrder.isDefined => fail
```

Reply:
BTW, I think the existing `functionWithUnsupportedSyntaxError` is fine here. We don't need to create a new error.

Reply:
> It only covers unsupported agg func. We should cover all cases in `validateFunction`

Thank you for the reminder.

Reply:
> BTW, I think the existing `functionWithUnsupportedSyntaxError` is fine here. We don't need to create a new error.

I tried to find out the suitable error, but it too many. Thank you for the reminder.